marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java

```diff
@@ -50,17 +50,48 @@ private int getBatchSize() {
 
 	@Override
 	public void call(InputStream[] input) {
-		call(null, null, null, input);
+		call(newCallContext(), input);
 	}
+
 	@Override
+	@Deprecated
 	public InputStream call(InputStream endpointState, SessionState session, InputStream workUnit, InputStream[] input) {
-		checkAllowedArgs(endpointState, session, workUnit);
-		return getCaller().arrayCall(getClient(), endpointState, session, workUnit, input);
+		newCallContext().withEndpointState(endpointState).withSessionState(session)
+				.withWorkUnit(workUnit);
+		return call(newCallContext(), input);
```

Comment:
The call on line 61 should use the call context constructed on line 59.

Currently, the constructed call context is thrown away, so the parameters aren't passed in the call.

The equivalent refactoring in InputOutputEndpointImpl is correct.